test/unit/math/require_util.hpp

```diff
@@ -89,39 +113,36 @@ template <template <class...> class checker, typename... Types>
 struct require_scal_checker {
   struct dummy {};
   static void unary() {
-    using stan::test::unary_require_tester;
     EXPECT_FALSE((unary_require_tester<checker, dummy>::value));
     EXPECT_TRUE((unary_require_tester<checker, Types...>::value));
   }
   static void not_unary() {
-    using stan::test::unary_require_tester;
-    EXPECT_TRUE((unary_require_tester<checker, dummy>::value));
-    EXPECT_FALSE((unary_require_tester<checker, Types...>::value));
+    EXPECT_TRUE((unary_not_require_tester<checker, dummy>::value));
+    EXPECT_FALSE((unary_not_require_tester<checker, Types...>::value));
```

Comment:
This is not part of the fix. I'm just strengthening this testcase. Basically:

```
  require_scal_checker<stan::require_not_double_or_int_t, double,
                       int>::not_unary();
```

Is saying: "test that require_not_double_or_int\<double\> disables the template and also test that require_not_double_or_int\<int\>" disables the template. not_unary() should fail the test if either enables the template. Previously, it only failed the test if both enabled the template, because unary_require_tester is built on a conjunction.